ZeldaFullEditor/ZSCustomOverworld.asm

```diff
@@ -2835,33 +3134,72 @@ ReplaceBGColor:
 {
     PHB : PHK : PLB
 
-    ; TODO: This may need to check if we are in a warp and then if so load the
-    ; custom color. If not, then chceck if its enabled or not.
-    ;SEP #$20 ; Set A in 8bit mode.
+    SEP #$20 ; Set A in 8bit mode.
 
-    ;LDA.w Pool_EnableBGColor : BNE .custom
-        ;REP #$20 ; Set A in 16bit mode.
+    LDA.w Pool_EnableBGColor : BNE .custom
+        REP #$20 ; Set A in 16bit mode.
 
-        ;PLB
+        PLB
 
-        ;RTL
+        RTL
 
-    ;.custom
+    .custom
 
-    ;REP #$20 ; Set A in 16bit mode.
+    REP #$20 ; Set A in 16bit mode.
 
-    LDA.b $8A : ASL : TAX ; Get area code and times it by 2.
-    LDA.w Pool_BGColorTable, X ; Get the color.
+    ; TODO: Saving the Y may not be necessary, done just in case.
```

Comment:
What's the TODO here?

Reply:
To investigate whether i do actually need to save the Y or not, Updated the comment to reflect that.